src/chainstate/stacks/miner.rs

```diff
@@ -1988,35 +1990,61 @@ impl StacksBlockBuilder {
 
                         if block_limit_hit == BlockLimitFunction::LIMIT_REACHED {
-                            return Ok(false);
+                            return Ok(None);
                         }
                         if get_epoch_time_ms() >= deadline {
                             debug!("Miner mining time exceeded ({} ms)", max_miner_time_ms);
-                            return Ok(false);
+                            return Ok(None);
                         }
 
                         // skip transactions early if we can
                         if considered.contains(&txinfo.tx.txid()) {
-                            return Ok(true);
+                            return Ok(Some(
+                                TransactionResult::skipped(
+                                    &txinfo.tx,
+                                    "Transaction already considered.".to_string(),
+                                )
+                                .convert_to_event(),
+                            ));
                         }
 
                         if let Some(nonce) = mined_origin_nonces.get(&txinfo.tx.origin_address()) {
                             if *nonce >= txinfo.tx.get_origin_nonce() {
-                                return Ok(true);
+                                return Ok(Some(
+                                    TransactionResult::skipped(
+                                        &txinfo.tx,
+                                        format!(
+                                            "Bad origin nonce, tx nonce {} versus {}.",
+                                            txinfo.tx.get_origin_nonce(),
+                                            *nonce
+                                        ),
+                                    )
+                                    .convert_to_event(),
+                                ));
                             }
                         }
                         if let Some(sponsor_addr) = txinfo.tx.sponsor_address() {
                             if let Some(nonce) = mined_sponsor_nonces.get(&sponsor_addr) {
                                 if let Some(sponsor_nonce) = txinfo.tx.get_sponsor_nonce() {
                                     if *nonce >= sponsor_nonce {
-                                        return Ok(true);
+                                        return Ok(Some(
+                                            TransactionResult::skipped(
+                                                &txinfo.tx,
+                                                format!(
+                                                    "Bad sponsor nonce, tx nonce {} versus {}.",
+                                                    sponsor_nonce, *nonce
+                                                ),
+                                            )
+                                            .convert_to_event(),
+                                        ));
```

Comment:
Do we actually want these events included in the event observer interface? These seems like they would be basically spurious event data. I'm not opposed to logging them, but re-broadcasting them over the event observer seems excessive.

Reply:
Something like option 3 -- they shouldn't be forwarded to the observer at all.

Reply:
I  tried it the reverting way.. but then the logic of skipping `Skipped` events was kind of dispersed.

So I pushed a version that puts this in `iterate_candidates`, that way we just have to write the logic once...

```
                    // Run `todo` on the transaction.
                    match todo(clarity_tx, &consider, self.cost_estimator.as_mut())? {
                        Some(tx_event) => {
                            match tx_event {
                                TransactionEvent::Skipped(_) => {
                                    // don't push `Skipped` events to the observer
                                }
                                _ => {
                                    output_events.push(tx_event);
                                }
                            }
                        }
                        None => {
                            debug!("Mempool iteration early exit from iterator");
                            break;
                        }
                    }
```